internal/websocket/ws.go

```diff
@@ -284,7 +289,21 @@ func (ws *Websocket) run(ctx context.Context) {
 			// Read loop
 			for {
 				var msg wrp.Message
+				ctx, cancel := context.WithTimeout(ctx, ws.inactivityTimeout)
 				typ, reader, err := ws.conn.Reader(ctx)
+
```

Comment:
I think the call to `cancel()` should be here since that immediately frees up the context.  Otherwise it will linger around until the inactivityTimeout finally comes into effect.